lib/beaker/options/validator.rb

```diff
@@ -0,0 +1,132 @@
+
+      # Raises an error if fail_mode is not a supported failure mode.
+      #
+      # @param [String] fail_mode Failure mode setting
+      # @return [nil] Does not return anything
+      def validate_fail_mode(fail_mode)
+        #check for valid fail mode
+        if fail_mode !~ VALID_FAIL_MODES
+          validator_error "--fail-mode must be one of fast or slow, not '#{fail_mode}'"
+        end
+      end
+
+      # Raises an error if hosts_setting is not a supported preserve hosts value.
+      #
+      # @param [String] hosts_setting Preserve hosts setting
+      # @return [nil] Does not return anything
+      def validate_preserve_hosts(hosts_setting)
+        #check for valid preserve_hosts option
+        if hosts_setting !~ VALID_PRESERVE_HOSTS
+          validator_error("--preserve_hosts must be one of always, onfail, onpass or never, not '#{hosts_setting}'")
+        end
+      end
+
+      # Raise an error if host does not have a platform defined.
+      #
+      # @param [::Beaker::Host] host A beaker host
+      # @param [String] name Host name
+      # @return [nil] Does not return anything
+      def validate_platform(host, name)
+        unless host['platform']
+          validator_error "Host #{name} does not have a platform specified"
+        end
+      end
+
+      # Raise an error if an item exists in both the include and exclude lists.
+      #
+      # @param [Array] include included items
+      # @param [Array] exclude excluded items
+      # @return [nil] Does not return anything
+      def validate_tags(include, exclude)
+
+        include.each do |included_tag|
+          # select items from exclude set that match included_tag
+          # no match is an empty list/array/[]
+          if exclude.select { |ex| ex == included_tag } != []
+            validator_error "tag '#{included_tag}' cannot be in both the included and excluded tag sets"
+          end
+        end
+
+      end
+
+      def validate_frictionless_roles(role_array)
+        if role_array.include?(FRICTIONLESS_ROLE) and !(role_array & FRICTIONLESS_ADDITIONAL_ROLES).empty?
+          validator_error "Only agent nodes may have the role 'frictionless'."
+        end
+      end
+
+      # Raise an error if the master count is incorrect.
+      #
+      # @param [Integer] count Count of roles with 'master'
+      # @return [nil] Nothing is returned
+      # @raise [ArgumentError] Raises if master count is greater than 1
+      def validate_master_count(count)
+        if count > 1
+          validator_error("Only one host/node may have the role 'master'.")
+        end
+
+      end
+
+      def validate_files(file_list, path)
+        if file_list.empty?
+          validator_error("No files found for path: '#{path}'")
+        end
+      end
+
+      def validate_path(path)
+        unless File.file?(path) || File.directory?(path)
+          validator_error("#{path} used as a file option but is not a file or directory!")
+        end
+      end
+
```

Comment:
Stylistic nit, I'm personally not a fan of the extra newlines before the `end`s at the end of files. I see it a lot here (and even more idiosyncratic newline applications in some other projects) and I'm not sure I understand it.